From bf2faa3bece53efdbd2c4d6ca8b478e3da3c171d Mon Sep 17 00:00:00 2001
From: Nebu Philips <nphilips@netlogicmicro.com>
Date: Tue, 24 Aug 2010 16:46:29 -0700
Subject: [PATCH 0636/1532] Added Tiny DTS File, now required by U-Boot.

This file holds info about NAE and FMN currently, but will support
additional interfaces such as DIMM EEPROM SPD etc.
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 dts/tiny.dts | 153 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 153 insertions(+)
 create mode 100644 dts/tiny.dts

diff --git a/dts/tiny.dts b/dts/tiny.dts
new file mode 100644
index 0000000..390541b
--- /dev/null
+++ b/dts/tiny.dts
@@ -0,0 +1,153 @@
+/* XLP8XX Device Tree Source 
+ *
+ */
+
+/dts-v1/;
+/ {
+	model = "MIPS,XLP8XX";
+	compatible = "NETL,XLP8XX_A0";
+	#address-cells = <1>;
+	#size-cells = <1>;
+	doms {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		dom@0 {
+			device_type = "domain";
+			heap-size = <0x0 0x4000000>;
+			shared-mem = <0x0 0x80000000 0x0 0x20000000>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			cpu {
+				onlinemask = <0xf>;
+				nae-rx-vc = <0>;
+				nae-fb-vc = <3>;
+			};
+			uart {
+				id = <0>;
+			};
+			memory {
+				reg = <0x00100000 0x0FF00000	// 255M at 1M
+					   0x20000000 0x20000000>;  // 512M at 512M
+			};
+			pic {
+			};
+		};
+	};
+
+	chosen {
+		bootargs = "xlp_noi2c console=ttyS0,115200 rdinit=/sbin/init";
+		nae-cfg {
+			num-ports = <4>;
+			start-port-id = <0>;
+			num-port-regs = <5>;
+			ucore {
+				src@1 {
+					num-opcodes = <52>;
+					mask = <0xffff>;
+					opcodes = <
+						0x3c1c0010 
+						0x279cf808 
+						0x3c020010 
+						0x2442f808 
+						0x3c030010 
+						0x2463f808 
+						0xac400000 
+						0x0043082b 
+						0x1420fffd 
+						0x24420004 
+						0x3c1d0010 
+						0x27bdf9c0 
+						0x27a50020 
+						0x27a60028 
+						0xaca00000 
+						0xacc00000 
+						0x0c000029 
+						0x00002021 
+						0x1000ffff 
+						0x00000000 
+						0x3c020000 
+						0x8c420104 
+						0x27bdffe0 
+						0x2403ffff 
+						0xafbf001c 
+						0xafb10018 
+						0x10430009 
+						0xafb00014 
+						0x3c100000 
+						0x26100104 
+						0x2411ffff 
+						0x0040f809 
+						0x2610fffc 
+						0x8e020000 
+						0x1451fffc 
+						0x00000000 
+						0x8fbf001c 
+						0x8fb10018 
+						0x8fb00014 
+						0x03e00008 
+						0x27bd0020 
+						0x00001821 
+						0x34088004 
+						0x8d060000 
+						0x30c5000f 
+						0x00052280 
+						0x34820001 
+						0x34088030 
+						0xad020000 
+						0x34088000 
+						0x0800002a 
+						0xad030000 
+					>;
+				};
+			};
+
+			port0 {
+				num-free-desc = <10>;
+				cfg-regs = <
+					0x35 0xd
+					0x1f 0x35
+					0x12 0x28
+					0x21 0x14
+					0x28 0x400
+				>;
+			};
+			port1 {
+				num-free-desc = <10>;
+				cfg-regs = <
+					0x35 0xd
+					0x1f 0x35
+					0x12 0x8
+					0x21 0x14
+					0x28 0x401
+				>;
+			};
+			port2 {
+				num-free-desc = <10>;
+				cfg-regs = <
+					0x35 0xd
+					0x1f 0x1f
+					0x12 0x8
+					0x21 0x14
+					0x28 0x402
+				>;
+			};
+			port3 {
+				num-free-desc = <10>;
+				cfg-regs = <
+					0x35 0xd
+					0x1f 0x12
+					0x12 0x8
+					0x21 0x14
+					0x28 0x403
+				>;
+			};
+		};
+		fmn{
+			spill-base-hi   = <0x00000000>;
+			spill-base-low  = <0xE0000000>;
+			spill-size  	= <0x1000000>;
+			tx-vc       	= <524>;
+			credit      	= <50>;
+		};
+	};
+};
-- 
1.9.1

